read-fonts/src/collections/int_set/bitpage.rs

```diff
@@ -97,18 +96,23 @@ impl BitPage {
     pub(crate) fn insert(&mut self, val: u32) -> bool {
         let ret = !self.contains(val);
         *self.element_mut(val) |= elem_index_bit_mask(val);
-        self.mark_dirty();
+        self.length += ret as u32;
         ret
     }
 
-    /// Marks `(val % page width)` a member of this set, but does not check if it was already a member.
+    /// Blindly marks `(val % page width)` a member of this set.
+    ///
+    /// This is used to maximize performance in cases where the return value
+    /// on [`insert()`] is not needed (such as for batching).
     ///
-    /// This is used to maximize performance in cases where the return value on [`insert()`] is not needed.
+    /// # WARNING:
+    /// After calling this method, the `length` value for this page is no longer
+    /// valid. The caller is responsible for ensuring that the
+    /// `recalculate_length` method is called before the
     ///
     /// [`insert()`]: Self::insert
-    pub(crate) fn insert_no_return(&mut self, val: u32) {
+    pub(crate) fn insert_no_return_and_promise_to_do_your_own_bookkeeping(&mut self, val: u32) {
```

Comment:
unclear to me if this is still worth it? It's only used for extending, and for small sets having to recompute the values of all pages is probably more expensive than just summing the returned bools 🤷

Reply:
This is a good question to answer via the benchmarks (https://github.com/googlefonts/fontations/tree/main/read-fonts/benches). They test across a variety of set sizes so you can see the exact tradeoffs being made. Recommend doing a before and after to see what happens.

Reply:
okay yea ripping all of this out is actually a win in many cases (larger sets especially) since it saves needing to update _all_ pages at the end.